plugin/pkg/scheduler/generic_scheduler.go

```diff
@@ -49,14 +50,24 @@ var ErrNoNodesAvailable = fmt.Errorf("no nodes available to schedule pods")
 func (f *FitError) Error() string {
 	var buf bytes.Buffer
 	buf.WriteString(fmt.Sprintf("pod (%s) failed to fit in any node\n", f.Pod.Name))
-	for node, predicates := range f.FailedPredicates {
-		reasons := make([]string, 0)
+	reasons := maps.NewCounter()
```

Comment:
- useful  for `Incr()` to happen in parallel though, i.e. multiple nodes incrementing.  Otherwise possibility of concurrent update panic i think
- the functionality / need to count bins is generalizable anyways .